core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -179,6 +179,7 @@ public final synchronized RunT newBuild() throws IOException {
         } catch (InstantiationException e) {
             throw new Error(e);
         } catch (IllegalAccessException e) {
+            LOGGER.log(Level.WARNING, String.format("A new build could not be created in job %s", asJob().getFullName()), e);
```

Comment:
+1 with @jtnord 

If the goal of that JIRA is to make issues more diagnosable, and as we're now Java 7+, we could do that instead:

```java
try {
    RunT lastBuild = getBuildClass().getConstructor(asJob().getClass()).newInstance(asJob());
    builds.put(lastBuild);
    lastBuild.getPreviousBuild(); // JENKINS-20662: create connection to previous build
    return lastBuild;
} catch (InstantiationException | IllegalAccessException | NoSuchMethodException e) {
    LOGGER.log(Level.WARNING, String.format("A new build could not be created in job %s", asJob().getFullName()), e);
    throw new Error(e);
} catch (InvocationTargetException e) {
    throw handleInvocationTargetException(e);
}
```

?